pkg/controller/source/source_test.go

```diff
@@ -203,4 +204,57 @@ var _ = Describe("Source", func() {
 			})
 		})
 	})
+	Describe("ChannelSource", func() {
```

Comment:
Add a test for when the ChannelSource is reused by multiple watches with different EventHandlers.